src/ICPlocalization.cpp

```diff
@@ -66,6 +66,27 @@ void ICPlocalization::setInitialPose(const Eigen::Vector3d &p,
 	tfPublisher_->setInitialPose(p, q);
 }
 
+void ICPlocalization::set2DPoseCallback(const geometry_msgs::PoseWithCovarianceStamped::ConstPtr& msg){
```

Comment:
Why not just use setInitialPose function that is already provided?

Reply:
I tried this at the beginning and this is what happen if I use this function:
https://drive.google.com/file/d/1HX2rv57EnuRgvH13XCRYDZr9nXWSQBvl/view?usp=sharing
You have to pause the bag in order to be able to set the initial position. I'm not sure that this will work on the real robot

Reply:
Ah yes. It's a thread synchronization problem. Okay, then leave it as it is.